File: leetcode/algorithms/476_number_complement/main.py
class NumberComplement:
    def findComplement(self, num: int) -> int:
        binary = "{0:b}".format(num)
        result = ""

        for i in binary:
            if i == "1":
                result += "0"
            else:
                result += "1"

        return int(result, 2)


    # Solution
    def solution(self, num: int) -> int:
        i = 1

        while i <= num:
            i = i << 1

        return (i - 1) ^ num

File: leetcode/algorithms/476_number_complement/main.java
class NumberComplement {
    public int findComplement(int num) {
        String[] binary = Integer.toBinaryString(num).split("");
        String result = "";

        for (String b : binary) {
            if (b.equals("1")) {
                result += "0";
            } else {
                result += "1";
            }
        }

        return Integer.parseInt(result, 2);
    }


    // Solution
    public int solution(int num) {
        int i = 0;
        int j = 0;

        while (i < num) {
            i += Math.pow(2, j);
            j++;
        }

        return i - num;
    }
}
